core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -230,6 +233,7 @@ public DefaultCacheManager(GlobalConfiguration globalConfiguration, Configuratio
       this.globalComponentRegistry.registerComponent(cacheDependencyGraph, CACHE_DEPENDENCY_GRAPH, false);
       this.globalComponentRegistry.registerComponent(authzHelper, AuthorizationHelper.class);
       this.stats = new CacheContainerStatsImpl(this);
+      this.health = HealthFactory.create(this);
```

Comment:
I'd be fine with a factory if `Health` was a component, even if it had only one implementation. But I don't like the idea of having several kind of factories in core.